app/services/determine_training_record_state_legacy.rb

```diff
@@ -243,7 +243,7 @@ def mentee_counts
       FROM "induction_records"
       WHERE
           "induction_records"."mentor_profile_id" = '#{participant_profile_id}'
-          AND ("induction_records"."induction_status" = 'active' OR "induction_records"."induction_status" = 'leaving')
+          AND ("induction_records"."induction_status" = 'active' OR ("induction_records"."induction_status" = 'leaving' AND "induction_records"."end_date" > CURRENT_TIMESTAMP))
```

Comment:
Is it worth adding a test cases for this?

We also have this logic in a few other places now as this is the legacy record state calculator that is in the process of being replaced. 

The new one calculates the mentoring state in two of places; the first is a concern that calculates the `current_mentees_count` on the initial query in the controller (which is an optimisation):

https://github.com/DFE-Digital/early-careers-framework/blob/0b65b163904c75a28dd8b4cede6ed345df3da94d/app/services/api/concerns/training_record_state_optimizable.rb#L22

The second place is in the`TrainingRecordState` model which serves as a fallback in case the concern has not been used:

https://github.com/DFE-Digital/early-careers-framework/blob/main/app/services/training_record_state.rb#L554

Reply:
Thanks @ethax-ross this came via a support ticket where a school is complaining that their mentor isn't mentoring but appears as if they are in their dashboard so I was working back from the view and wasn't aware this logic was elsewhere.